game-core/src/main/java/games/strategy/util/IntegerMap.java

```diff
@@ -19,15 +19,15 @@
 
   /** Creates new IntegerMap. */
   public IntegerMap() {
-    mapValues = new HashMap<>();
+    mapValues = new LinkedHashMap<>();
```

Comment:
Just noting that now there's basically no difference between `IntegerMap` and `LinkedIntegerMap`.  I'm assuming, due to compatibility issues, we can't simply change the `IntegerMap` instance that needs to be ordered to `LinkedIntegerMap`?

Reply:
So I'd be interested in thoughts on performance impact of LinkedHashMap vs HashMap. If you query the usage of LinkedIntegerMap and IntegerMap, there are only 24 references to LinkedIntegerMap while there are 957 references to IntegerMap. I have a pretty strong feeling that a good portion of those should be using an ordered key set. My preference would be to consider just removing LinkedIntegerMap when we do an incompatible release. I think that is probably the easiest way forward unless we feel the performance considerations are that impactful.

Reply:
I have no evidence to back this up, but my guess is our `IntegerMap`s aren't big enough to cause a noticeable performance difference between the two implementations.

However, of more concern, is if switching `IntegerMap` from an unordered collection to an ordered one is going to have some functional side effect on those 957 references?  Again, my guess is probably not, but I'd like to make myself feel better about that before just blowing it off. :smile:

(Maybe something to add to the next incompatible release wishlist is to figure out a better way to "preserve ordering" from the map XML.  Using a hidden implementation detail like this seems brittle.  We need something like a separate, but parallel, domain object that represents the ordering of these types of collections.  That way we're free to use whatever collection implementation we wish, and when it finally gets to the UI, we can use the parallel domain object to enforce the desired ordering just-in-time.)

Reply:
@ssoloff Yeah, I couldn't think of any reasons enforcing order in the IntegerMap would have negative side effects but its theoretically possible. I think I'm alright with that very minimal risk. I created an issue to discuss this further and what we'd like to do for an incompatible release: #3122